#!/usr/bin/env python

PACKAGE='socialspace_navigation_layer'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("enabled",       bool_t,   0, "Whether to apply this plugin",                              True)
gen.add("cutoff",        double_t, 0, "Smallest value to be published on the costmap",             10.0, 1.0, 254.0)
gen.add("amplitude",     double_t, 0, "Amplitude of costs at peak",                                77.0, 1.0, 254.0)
gen.add("robot_bounds",  double_t, 0, "Area around the robot to consider when updating the costs",  2.0, 0.0)

gen.add("personal_sigma_h_factor", double_t, 0, "sigma_h in asymmetric Gaussian for modelling passing space", 1.0, 1.0)

gen.add("passing_right_hand_traffic", bool_t,   0, "Whether the traffic system is a right hand one.",            True)
gen.add("passing_sigma_h",            double_t, 0, "sigma_h in asymmetric Gaussian for modelling passing space", 2.0,  0.0)
gen.add("passing_sigma_s",            double_t, 0, "sigma_s in asymmetric Gaussian for modelling passing space", 0.25, 0.0)
gen.add("passing_sigma_r",            double_t, 0, "sigma_r in asymmetric Gaussian for modelling passing space", 0.01, 0.0)

gen.add("activity_object_sigma", double_t, 0, "sigma used for smoothing the activity/affordance space with objects", 0.15, 0.0)
gen.add("activity_human_sigma",  double_t, 0, "sigma used for smoothing the activity/affordance space with humans",  0.5, 0.0)

exit(gen.generate(PACKAGE, "socialspace_navigation_layer", "SocialspaceLayer"))
